src/main/java/me/hypherionmc/sdlinklib/discord/commands/WhitelistCommand.java

```diff
@@ -173,16 +172,14 @@ protected void execute(CommandEvent event) {
                         if (!tables.isEmpty()) {
                             tables.forEach(SQLiteTable::delete);
 
-                            String nickName = (event.getMember().getNickname() == null || event.getMember().getNickname().isEmpty()) ? event.getAuthor().getName() : event.getMember().getNickname();
-                            if (pattern.matcher(nickName).matches()) {
-                                nickName = pattern.matcher(nickName).replaceAll("");
-                            }
-
-                            try {
-                                event.getMember().modifyNickname(nickName).queue();
-                            } catch (Exception e) {
-                                if (modConfig.generalConfig.debugging) {
-                                    e.printStackTrace();
+                            String nickName = event.getMember().getEffectiveName();
+                            if (pattern.matcher(nickName).find()) {
+                                try {
+                                    event.getMember().modifyNickname(null).queue();
```

Comment:
Same as before, don't clear the nickname

Reply:
Same as before